build.go

```diff
@@ -31,6 +34,39 @@ func Build(ctx context.Context, c client.Client) (*client.Result, error) {
 		}
 	}
 
+	var (
+		manifest llb.State
+		src      llb.State
+	)
+
+	// Use frontend input instead of `llb.Local` if supported by the gateway capabilities.
+	if (&gwcaps).Supports(gwpb.CapFrontendInputs) == nil {
+		inputs, err := c.Inputs(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		inputManifest, ok := inputs[InputNameManifest]
+		if ok {
```

Comment:
if either of these cases, if input is not found it should default to `llb.Local`. Even if `CapFrontendInputs` is supported.

Reply:
@hinshun looks like this needs addressing if you still want to get this merged